src/query/v2/plan/scoped_profile.hpp

```diff
@@ -13,13 +13,47 @@
 
 #include <cstdint>
 
+#include <json/json.hpp>
+
 #include "query/v2/context.hpp"
 #include "query/v2/plan/profile.hpp"
 #include "utils/likely.hpp"
 #include "utils/tsc.hpp"
 
 namespace memgraph::query::v2::plan {
 
+class ScopedCustomProfile {
+ public:
+  explicit ScopedCustomProfile(const std::string_view custom_data_name, ExecutionContext &context)
+      : custom_data_name_(custom_data_name), start_time_{utils::ReadTSC()}, context_{&context} {}
+
+  ScopedCustomProfile(const ScopedCustomProfile &) = delete;
+  ScopedCustomProfile(ScopedCustomProfile &&) = delete;
+  ScopedCustomProfile &operator=(const ScopedCustomProfile &) = delete;
+  ScopedCustomProfile &operator=(ScopedCustomProfile &&) = delete;
+
+  // If an exception is thrown in any of these functions that signals a problem that is much bigger than we could handle
+  // it here, thus we don't attempt to handle it.
+  // NOLINTNEXTLINE(bugprone-exception-escape)
+  ~ScopedCustomProfile() {
+    if (nullptr != context_->stats_root) {
+      auto &custom_data = context_->stats_root->custom_data[custom_data_name_];
```

Comment:
How is the presence of `custom_data_name_custom_data_name_ guaranteed in `custom_data`?

Reply:
[`operator[]` works similarly to `std::map`'s implementation](https://json.nlohmann.me/features/element_access/unchecked_access/#overview): adds a default value (`Null` in this case) if the key is missing.

Reply:
Ah nice, I did not know that